ast/capabilities_test.go

```diff
@@ -181,14 +181,33 @@ func TestCapabilitiesMinimumCompatibleVersion(t *testing.T) {
 			version: "0.38.0",
 		},
 		{
-			note: "features",
+			note: "features (string prefix ref)",
 			module: `
 				package x
 				import future.keywords.if
 				p.a.b.c.d if { true }
 			`,
 			version: "0.46.0",
 		},
+		{
+			note: "features (general ref)",
+			module: `
+				package x
+				import future.keywords.if
+				p.a.b[c].d if { c := "foo" }
+			`,
+			version: "0.59.0",
+		},
+		{
+			note: "features (general ref + string prefix ref)",
+			module: `
+				package x
+				import future.keywords.if
+				p.a.b.c.d if { true }
+				p.a.b[c].d if { c := "foo" }
+			`,
+			version: "0.59.0",
+		},
```

Comment:
I've added support for the `rule_head_refs` capability feature to the minimum compatible version calculation @tsandall .

How do we envision this working when we start removing capabilities, such as the deprecated built-ins?

Then, the following policy 

```rego
package x
p { all([true, false]) }
```

will still output `0.17.0` as minimum version, but it won't be compatible with `1.0`.
Will we just add a `MaximumCompatibleVersion()`, that'll output whatever the last supported version is for the capability?

Reply:
I hadn't thought about the deprecated case. I wonder if a better interface would be to return the compatible version range?

Reply:
Yes, that makes sense 👍 .
I wonder if this change fits better as part of https://github.com/open-policy-agent/opa/issues/6361, though, as it wouldn't make much difference here.